src/utils/entityNameGenerators.ts

```diff
@@ -0,0 +1,12 @@
+interface IGenerateScmRepositoryName {
+  internalUrl: string;
+}
+
+/**
+ * Parses and generate SCM Repository name from internalUrl.
+ *
+ * @param object - object containing internalUrl field
+ * @returns  SCM Repository name
+ */
+export const generateScmRepositoryName = ({ internalUrl }: IGenerateScmRepositoryName) =>
```

Comment:
In this case I would specify also return type. All generator functions should return the same type which is `string`, ideally something like `IEntityNameGenerator` would be created.